tethys_apps/base/workspace.py

```diff
@@ -254,6 +256,42 @@ def a_controller(request):
     return TethysWorkspace(workspace_directory)
 
 
+def get_user_workspace(request_or_user) -> TethysWorkspace:
+    """
+    Get the dedicated workspace directory for active user of request given or the given user.
+
+    Args:
+        request_or_user: Either an HttpRequest with active user session or Django User object.
+    
+    Raises:
+        ValueError: if request_or_user is not of type HttpRequest or User.
+        AssertionError: if quota for the user workspace has been exceeded.
+
+    Returns:
+        TethysWorkspace: workspace object bound to the user's workspace directory.    
+    """
+    from django.contrib.auth.models import User
+    from tethys_apps.utilities import get_active_app
+
+    # Get user
+    if isinstance(request_or_user, HttpRequest):
+        user = request_or_user.user
+    elif isinstance(request_or_user, User):
+        user = request_or_user
+    else:
+        raise ValueError(f'Argument "request_or_user" must be of type HttpRequest or User: '
+                         f'"{type(request_or_user)}" given.')
+
+    assert passes_quota(user, 'user_workspace_quota')
+
+    # Get the active app
+    app = get_active_app(request_or_user, get_class=True)
```

Comment:
Have you tested the behavior if a User is passed? I'm not sure get_active_app accept anything other than a request since it needs the url to figure out what the active app is.

Reply:
Good catch. I've refactored this and tested it.